files/var/www/ks/bootstrap_simp_client

```diff
@@ -0,0 +1,564 @@
+  end
+
+  def set_up_log
+    log_file = @options[:log_file]
+    if File.exist?(log_file)
+      timestamp = File.stat(log_file).mtime.strftime("%Y-%m-%dT%H%M%S")
+      log_backup = "#{log_file}.#{timestamp}"
+      debug("Backing up #{log_file} to #{log_backup}")
+      FileUtils.mv(log_file, log_backup)
+    end
+
+    # Note: If the ownership of this file is root:root, puppet agent
+    # will complain in an 'err'-level message, but not fail, when it
+    # writes to the file. However, since there is no puppet user
+    # on the client, this can't be fixed...
+    FileUtils.touch(log_file)
+  end
+
+  # raises ConfigurationError upon validation failure
+  def validate_options
+    unless not_nil_or_empty?(@options[:puppet_server])
+      raise ConfigurationError.new('No Puppet server specified')
+    end
+
+    unless not_nil_or_empty?(@options[:puppet_ca])
+      raise ConfigurationError.new('No Puppet CA specified')
+    end
+
+    if (@options[:puppet_ca_port] < 1) or (@options[:puppet_ca_port] > 65535)
+      msg = "Invalid Puppet CA port '#{@options[:puppet_ca_port]}': must be in [0,65535]"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:num_puppet_runs] > 0
+      msg = "Invalid number of puppet agent runs '#{@options[:num_puppet_runs]}': must be > 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:puppet_keylength] > 0
+      msg = "Invalid Puppet keylength '#{@options[:puppet_keylength]}': must be > 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:puppet_wait_for_cert] >= 0
+      msg = "Invalid Puppet wait for cert '#{@options[:puppet_wait_for_cert]}': must be >= 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:initial_retry_interval] > 0
+      msg = "Invalid initial retry interval '#{@options[:initial_retry_interval]}': must be > 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:retry_factor] > 0
+      msg = "Invalid retry factor '#{@options[:retry_factor]}': must be > 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    unless @options[:max_seconds] > 0
+      msg = "Invalid max seconds '#{@options[:max_seconds]}': must be > 0"
+      raise ConfigurationError.new(msg)
+    end
+
+    if @options[:quiet] and @options[:debug]
+      raise ConfigurationError.new('--quiet and --debug cannot be used together')
+    end
+  end
+
+  ############# logging methods #############
+  def title(msg, level)
+    case level
+    when 1
+       "*** #{msg} ***"
+    when 2
+       "------ #{msg} ------"
+    when 3
+       "......... #{msg} ........."
+    else
+       msg
+    end
+  end
+
+  def error(msg)
+   log(:err, msg, true)
+  end
+
+  def warn(msg)
+   log(:warning, msg, true)
+  end
+
+  def info(msg='')
+   log(:info, msg, false)
+  end
+
+  def debug(msg='')
+   log(:debug, msg, false)
+  end
+
+  def log(level, msg, suppress_empty_msgs)
```

Comment:
Since you have this method, do you want to go ahead and output to syslog at the same time?

```
require 'syslog/logger'

syslog = Syslog::Logger.new('simp_bootstrap')

log.error(whatever)
log.info(whatever)
log.warning(whatever)
```

Reply:
This will only capture the messages simp_bootstrap_client logs, not the messages logged by puppet agent or fixfiles.  Is this still useful without the puppet agent logs?

Reply:
I think so. It could actually be a great way to figure out the basics of why something failed without having to hit the host if syslog is enabled at kickstart time.

Reply:
For CentOS7, the messages from bootstrap_simp_client will be automatically sent to syslog by systemd. So, I don't think this work is necessary.

Reply:
:+1: